packages/prisma/schema.prisma

```diff
@@ -464,6 +464,7 @@ model SelectedCalendar {
   externalId   String
   credential   Credential? @relation(fields: [credentialId], references: [id], onDelete: Cascade)
   credentialId Int?
+  metadata     Json?
```

Comment:
`comment-blocking`: I believe we should avoid doing `Json` metadata. Not only does it complicate maintenance as we have no clear schema of how it works, but it also does not help Prisma to generate good/optimized queries.

Discussed this with @keithwillcode and we believe it's better to add specific columns to what is required (for now only google, but for future use cases we should prefix column names with the calendar type eg: `google_calendarId` etc).

Adding Indexes and uniqueness to those columns will also greatly improve performance.

Reply:
What if we treat each event coming in as a "Booking" in a sense? Are we only concerned that the event belongs to the user, is associated with the selected calendar, the start time, and end time?

I think this way avoids having column names associated with specific calendars and makes it easier to add more calendars to the cache.

Reply:
@joeauyeung  that would be too much data to handle IMO. We're just trying to cache prdefined api call responses. And since those can vary a lot. That's why I chose JSON. Since we aren't planning to query on it. We can just parse it

Reply:
Then we would need to spend more resources ensuring data is always fresh and prevent staleness. Which defeats the purpose of a quick and easy way to achieve most of the benefit.

Reply:
I think we're already ensuring that the data is fresh anyway. When a user makes changes to their calendar, we store the response that is sent to our cache.

That means we have to process the response when someone grabs the availability. My thought process is let's do most of the computation before a user visits the booking page.

Reply:
That's planned for v2

Reply:
> That's planned for v2

Ok awesome. I was thinking about @exception's comment about how to avoid using JSON data and seeing if we could apply it to this PR.